torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -68,6 +68,62 @@ c10::optional<at::ScalarType> PromoteIntegralType(
                                                                    : opt_dtype;
 }
 
+bool IsTypeWithLargerRangeThanLong(torch::ScalarType dtype) {
+  return dtype == at::ScalarType::BFloat16 || dtype == at::ScalarType::Float ||
+         dtype == at::ScalarType::Double;
+}
+
+// Return the upper limit for a given type. For floating point typesreturn
+// 2^mantissa to ensure that every value is representable.
+int64_t GetIntegerUpperLimitForType(torch::ScalarType dtype) {
```

Comment:
I'd rather not to copy the bulk of the `XlaHelpers::MinMaxValues()` API here.
Just use the 2^MANTISSA special case for needing types, and use `XlaHelpers::MinMaxValues()` API for the others.